### PR TITLE
debugging: add log4j failure and failover

### PR DESCRIPTION
**Pull request by PettitWesley**
Signed-off-by: Wesley Pettit <wppttt@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by lubingfeng — APPROVED** [review]

File: troubleshooting/debugging.md [hunk 1]
```diff
@@ -161,6 +165,57 @@ And Fluent Bit logs show this SIGTERM message, then this indicates that Fluent B
...
+
+While we can not prevent connection failures from occurring, we have found several mitigations.
+
+##### Mitigation 1: Enable Workers for all outputs
+
+AWS testing has found that TCP input throughput is maximized when Fluent Bit is performing well. If you enable at least one worker for all outputs, this ensures your outputs each have their own thread(s). This frees up the main Fluent Bit scheduler/input thread to focus on the inputs. 
+
+##### Mitigation 2: Log4J Failover to STDOUT with Appender Pattern
+
+If the TCP plugin fails for fluent bit, you can configure a failover appender. The Log4J XML below configures logs to be retried to STDOUT failover. It is difficult to search stdout for these failed logs if/because STDOUT is cluttered with other non log4j logs. You might prefer to have a special pattern for logs that failover and go to stdout:
```
Comment (matthewfala) [inline troubleshooting/debugging.md:196]: Feel free to ignore. Fluent Bit here is not capitalized but it is capitalized everywhere else

File: troubleshooting/debugging.md [hunk 2]
```diff
@@ -161,6 +165,57 @@ And Fluent Bit logs show this SIGTERM message, then this indicates that Fluent B
...
+
+If this occurs, your first step is to check if it is simply a misconfiguration. Ensure you have configured a Fluent Bit TCP input for the same port that log4j is using. 
+
+Unfortunately, AWS has received occasional reports that log4j TCP appender will fail randomly in a previously working setup. AWS is tracking reports of this problem here: https://github.com/aws/aws-for-fluent-bit/issues/294
+
+While we can not prevent connection failures from occurring, we have found several mitigations.
+
+##### Mitigation 1: Enable Workers for all outputs
+
+AWS testing has found that TCP input throughput is maximized when Fluent Bit is performing well. If you enable at least one worker for all outputs, this ensures your outputs each have their own thread(s). This frees up the main Fluent Bit scheduler/input thread to focus on the inputs.
```
Comment (matthewfala) [inline troubleshooting/debugging.md:192]: This makes sense. Good solution.

**Review by DrewZhang13 — APPROVED** [review]

File: troubleshooting/debugging.md (hunk 2 above)
Comment (matthewfala) [reply troubleshooting/debugging.md:192]: Do you think you should mention that workers is set to 1 by default so people don't get confused?

**Review by matthewfala — APPROVED** [review]

File: troubleshooting/debugging.md (hunk 1 above)
Comment (PettitWesley) [reply troubleshooting/debugging.md:196]: thanks

File: troubleshooting/debugging.md (hunk 2 above)
Comment (PettitWesley) [reply troubleshooting/debugging.md:192]: yea

**Pull request merged**